src/signin.js

```diff
@@ -888,30 +894,48 @@ function loginStatusBox (dom, listener) {
       }
     }
     var signOutButton = dom.createElement('input')
-    signOutButton.className = 'WebIDCancelButton'
+    // signOutButton.className = 'WebIDCancelButton'
     signOutButton.setAttribute('type', 'button')
     signOutButton.setAttribute('value', logoutLabel)
-    signOutButton.addEventListener('click', zapIt, false)
+    signOutButton.setAttribute('style',  signInButtonStyle + 'background-color: #eee;')
+    signOutButton.addEventListener('click', logoutButtonHandler, false)
     return signOutButton
   }
 
   box.refresh = function () {
-    let me = defaultTestUser()
-    let meUri = me ? me.uri : ''
-    if (box.me !== meUri) {
-      widgets.clearElement(box)
-      if (me) {
-        box.appendChild(logoutButton(me))
+    solidAuthClient.currentSession().then(session => {
+      if (session && session.webId) {
+        me = $rdf.sym(session.webId)
       } else {
-        box.appendChild(signInOrSignUpBox(dom, setIt))
+        me = null
       }
-    }
-    box.me = meUri
+      if ((me && box.me !== me.uri) || (!me && box.me)) {
+        widgets.clearElement(box)
+        if (me) {
+          box.appendChild(logoutButton(me))
+        } else {
+          box.appendChild(signInOrSignUpBox(dom, setIt))
+        }
+      }
+      box.me = me ? me.uri : null
+    }, err => {
+      alert('loginStatusBox: ' + err)
+    })
+  }
+
+  if (solidAuthClient.trackSession) {
+    solidAuthClient.trackSession(session => {
+      if (session && session.webId) {
+        me = $rdf.sym(session.webId)
+      } else {
+        me = null
+      }
+      box.refresh()
+    })
   }
```

Comment:
My only worry here is whether you will not end up with a double-refresh, i.e., should the other refresh code be put in the `else` statement? (i.e., trackSession does not exist)